lib/trivial.nix

```diff
@@ -58,9 +58,8 @@ rec {
      of the next function, and the last function returns the
      final value.
   */
-  pipe = val: functions:
-    let reverseApply = x: f: f x;
-    in builtins.foldl' reverseApply val functions;
```

Comment:
I initially moved the `let in` outside the arguments because it doesn't depend on them, which also makes it faster, after which it's
```nix
let reverseApply = x: f: f x;
in val: functions: builtins.foldl' reverseApply val functions
```

And with this we can save the allocation of two lambdas by just using the partially applied `builtins.foldl' reverseApply`, also making it faster if anything.

The type is documented in the docstring, so people shouldn't depend on the implementation for knowing this.

I think it should be fine to do here, it's just a minor improvement I noticed when wanting to use this function.